code/procs/helpers.dm

```diff
@@ -219,27 +219,24 @@ proc/get_angle(atom/a, atom/b)
 		if(curr.density) return 0
 	return 1
 
-/proc/do_mob(var/mob/user , var/atom/target as turf|obj|mob, var/time = 30) //This is quite an ugly solution but i refuse to use the old request system.
+/proc/do_mob(mob/user , atom/target as turf|obj|mob, time = 30) //This is quite an ugly solution but i refuse to use the old request system.
 	if(!user || !target) return 0
 	. = 0
-	var/user_loc = user.loc
-	var/target_loc = target.loc
-	var/holding = user.equipped()
-	sleep(time)
-	if (!user || !target)
-		return 0
-	if ( user.loc == user_loc && target.loc == target_loc && user.equipped() == holding && !is_incapacitated(user) && !user.lying )
-		return 1
+	var/datum/action/bar/icon/mob_doer/doer = new(user, target, time)
+	actions.start(doer, user)
+	while (doer.success == -1)
+		sleep(1)
```

Comment:
why sleep and not lagcheck?
applies for both.

Reply:
It's just trying to wait until the success variable changes. Lagcheck would only sleep if the server was lagging, causing non-lagging servers to go into an infinite loop.

Reply:
fair, this just sucks but it's not your fault